libsolidity/ast/Types.cpp

```diff
@@ -1378,12 +1378,25 @@ bool StringLiteralType::operator==(Type const& _other) const
 
 std::string StringLiteralType::toString(bool) const
 {
-	size_t invalidSequence;
+	auto isPrintableASCII = [](const string& s)
+	{
+		for (auto c: s)
+		{
+			if (static_cast<unsigned>(c) <= 0x1f || static_cast<unsigned>(c) >= 0x7f)
+				return false;
+		}
+		return true;
+	};
+
+	string ret = isPrintableASCII(m_value) ?
+		("literal_string \"" + m_value + "\"") :
+		("literal_string hex\"" + util::toHex(util::asBytes(m_value)) + "\"");
 
+	size_t invalidSequence;
 	if (!util::validateUTF8(m_value, invalidSequence))
-		return "literal_string (contains invalid UTF-8 sequence at position " + util::toString(invalidSequence) + ")";
+		ret += " (contains invalid UTF-8 sequence at position " + util::toString(invalidSequence) + ")";
```

Comment:
Still not happy having this, but this is the only error message indication we have.

Reply:
Tried changing this and unfortunately there are 3 different places in the typechecker this would need to be added 😢

Reply:
I did review before reading this. You can ignore it for now

Reply:
Got rid of this.